vcd/resource_vcd_nsxt_edgegateway_dhcp_forwarding.go

```diff
@@ -0,0 +1,205 @@
+}
+
+func resourceVcdNsxtEdgegatewayDhcpForwardingRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return genericVcdNsxtEdgegatewayDhcpForwardingRead(ctx, d, meta, "resource")
+}
+
+func genericVcdNsxtEdgegatewayDhcpForwardingRead(_ context.Context, d *schema.ResourceData, meta interface{}, origin string) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	orgName := d.Get("org").(string)
+	edgeGatewayId := d.Get("edge_gateway_id").(string)
+
+	nsxtEdge, err := vcdClient.GetNsxtEdgeGatewayById(orgName, edgeGatewayId)
+	if govcd.ContainsNotFound(err) {
+		// When parent Edge Gateway is not found - this resource is also not found and should be
+		// removed from state
+		d.SetId("")
+		if origin == "datasource" {
+			return diag.Errorf("[DHCP forwarding DS read] error retrieving NSX-T Edge Gateway DHCP forwarding: %s", err)
+		}
+		log.Printf("[DEBUG] Edge gateway no longer exists. Removing from tfstate")
+		return nil
+	}
+
+	if err != nil {
+		return diag.Errorf("[DHCP forwarding read] error: %s", err)
+	}
+
+	dhcpForwardConfig, err := nsxtEdge.GetDhcpForwarder()
+	if err != nil {
+		return diag.Errorf("[DHCP forwarding read] error retrieving NSX-T Edge Gateway DHCP forwarding: %s", err)
+	}
+
+	// DHCP forwarding does not have its own ID - it is a part of Edge Gateway
+	d.SetId(edgeGatewayId)
+	dSet(d, "enabled", dhcpForwardConfig.Enabled)
+
+	err = d.Set("dhcp_servers", convertStringsToTypeSet(dhcpForwardConfig.DhcpServers))
+	if err != nil {
+		return diag.Errorf("error setting dhcp_servers attribute: %s", err)
+	}
+
+	if !dhcpForwardConfig.Enabled {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Warning,
+				Summary:  "DHCP forwarding IP addresses will not be changed if the service is disabled",
+			},
+		}
+	}
+
+	return nil
+}
+
+func resourceVcdNsxtEdgegatewayDhcpForwardingDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	unlock, err := vcdClient.lockParentVdcGroupOrEdgeGateway(d)
+	if err != nil {
+		return diag.Errorf("[DHCP forwarding delete] %s", err)
+	}
+
+	defer unlock()
+
+	orgName := d.Get("org").(string)
+	edgeGatewayId := d.Get("edge_gateway_id").(string)
+
+	nsxtEdge, err := vcdClient.GetNsxtEdgeGatewayById(orgName, edgeGatewayId)
+	if err != nil {
+		return diag.Errorf("[DHCP forwarding delete] error retrieving Edge Gateway: %s", err)
+	}
+
+	// There is no "delete" for DHCP forwarding. It can only be updated to empty values (disabled)
+	_, err = nsxtEdge.UpdateDhcpForwarder(&types.NsxtEdgeGatewayDhcpForwarder{})
+	if err != nil {
+		return diag.Errorf("[DHCP forwarding delete] error updating DHCP forwarding: %s", err)
+	}
+
+	return nil
+}
+
+// resourceVcdNsxtEdgegatewayDhcpForwardingImport imports DHCP forwarding configuration for NSX-T
+// Edge Gateway.
+// The import path for this resource is Edge Gateway. ID of the field is also Edge Gateway ID as
```

Comment:
Is this an incomplete comment? _"ID of the field is also Edge Gateway ID as..."_

Reply:
I like to have my lines short, so the rest of the comment is on the next line

Reply:
same as the previous, the comment is finished on the next line

Reply:
Ah, sorry, I was using the split view and I didn't see it 😇

Reply:
Ah, sorry, I was using the split view and I didn't see it 😇